### ISSUE TITLE
CMake error when the source directory full path contains "C++/"

### ISSUE THREAD

**Issue by gdamiand** [CGAL/cgal#7866]
_Please use the following template to help us solving your issue._

## Issue Details

Error when running cmake, with examples ON.

## Source Code

cgal master

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Ubuntu 20.04.6 LTS
* CGAL version: master branch
* cmake version: 3.27.8

**Comment by gdamiand** [comment]
Error logs

[err2.log](https://github.com/CGAL/cgal/files/13388648/err2.log)

**Comment by afabri** [comment]
Here is the first error: 
````
== Generating build files for examples ==
CMake Error at Installation/cmake/modules/CGAL_Macros.cmake:587 (string):
  string sub-command REGEX, mode REPLACE failed to compile regex
  "/home/to/dev/C++/cgal2/.*/examples/".
Call Stack (most recent call first):
  Installation/examples/CMakeLists.txt:32 (process_cgal_subdirectory)

````

**Comment by lrineau** [comment]
That is because of `C++` in the path to the repository. The character `+` is special in a regular expression, and `C++` means:
  - `C+` (any strictly positive number of `C`)
  - and then `+` is a syntax error.

I will try to fix that. Meanwhile, just rename the directory `C++`, or move the `cgal` repository elsewhere.

**Comment by gdamiand** [comment]
Thanks Laurent !

**Comment by lrineau** [comment]
Should be fixed by the PR https://github.com/CGAL/cgal/pull/7869. I have based the patch on the branch `5.5.x-branch`. That bug is here since the introduction of CMake in CGAL (version 3.4 in 2009).

**Comment by lrineau** [comment]
This issue has been solved by the merge of PR #7869 in `master` and release branches.
